todoapp/src/components/TodoApp.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+  <DeletePopup
+    v-if="isPopupDisplayed"
+    :is-pop-up="isPopUp"
+    :selected-todo-id="selectedTodoId"
+    :is-finished-todo-pop-up="isFinishedTodoPopUp"
+    :selected-finished-todo-id="selectedFinishedTodoId"
+    @remove-task="removeTask"
+    @close-pop-up="closePopUp"
+  />
+  <div
+    class="relative flex flex-col items-center mt-32 max-w-288 w-4/5 xsm:max-w-500 xsm:w-11/12 xsm:mt-44 md:max-w-610 md:w-4/5 md:mt-48"
+  >
+    <Header @add-todo="addTodo" />
+    <SearchTodos
+      v-if="todos.length"
+      v-model="localSearchInput"
+      @clearSearchInput="clearSearchInput"
+      @doSearch="doSearch"
+    />
+    <FilterTodos v-if="todos.length" :selectedSortingButtons="selectedSortingButtons" />
+    <div class="flex flex-col w-full gap-7 xsm:gap-10 md:gap-12">
+      <Todos
+        :todos="todos"
+        :search-input="searchInput"
+        :search-todos="searchTodos"
+        :current-edited-todo="currentEditedTodo"
+        :selected-sorting-buttons="selectedSortingButtons"
+        @get-todo-id="getTodoId"
+        @on-todo-update="updateTodo"
+        @display-popup="displayPopup"
+        @add-search-todo="addSearchTodo"
+        @mark-todo-status="markTodoStatus"
+      />
+      <FinishedTodos
+        v-if="finishedTodos.length"
+        :search-input="searchInput"
+        :finished-todos="finishedTodos"
+        :current-edited-todo="currentEditedTodo"
+        :search-finished-todos="searchFinishedTodos"
+        :selected-sorting-buttons="selectedSortingButtons"
+        @get-todo-id="getTodoId"
+        @on-todo-update="updateTodo"
+        @display-popup="displayPopup"
+        @mark-todo-status="markTodoStatus"
+      />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref, computed } from 'vue'
+import moment from 'moment'
+import type { TodoType } from '../types/text'
+import Header from './Header.vue'
+import Todos from './Todos/Todos.vue'
+import FinishedTodos from './FinishedTodos/FinishedTodos.vue'
+import SearchTodos from './SearchTodos.vue'
+import FilterTodos from './FilterTodos.vue'
+import DeletePopup from './DeletePopup.vue'
+
+const todos = ref<TodoType[]>([])
+const finishedTodos = ref<TodoType[]>([])
+const isPopUp = ref(false)
+const isFinishedTodoPopUp = ref(false)
+const selectedTodoId = ref()
+const selectedFinishedTodoId = ref()
+const searchInput = ref('')
+const localSearchInput = ref('')
+const currentEditedTodo = ref<number | null>(null)
+
+const selectedSortingButtons = ref({
+  title: { selected: true, order: true },
+  description: { selected: false, order: true },
+  priority: { selected: false, order: true },
+  date: { selected: false, order: true }
+})
+
+const commonTodoProperties = {
+  text: 'Add a description',
+  priority: 0,
+  isPriorityChange: false,
+  isChecked: false,
+  isEditing: false
+}
+
+function addTodo() {
```

Comment:
The `addTodo` and `addTodo` functions are almost the same. Please create general function for the common code s.